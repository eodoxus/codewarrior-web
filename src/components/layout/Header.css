.header {
  border-bottom: 1px solid #cfcfcf;
  color: #fff;
  background-color: #2c3e50;
  display: flex;
}

.selfie {
  flex: 1;
  display: flex;
  justify-content: flex-end;
  margin-right: 20px;
}

.selfieFrame {
  width: 75px;
  height: 75px;
  background: #fff;
  padding: 3px;
  cursor: pointer;
  border: 1px solid #dce4ec;
  border-radius: 4px;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.075) inset;
  transition: border-color 0.15s ease-in-out 0s, box-shadow 0.15s;
  position: relative;
  top: 10px;
}

.selfie img {
  width: 100%;
}

.title {
  flex: 1;
  font-size: 170%;
  margin: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.title .subtitle {
  font-size: 50%;
  margin: 0;
}

.contact {
  display: flex;
  flex: 2;
}

.contactIcons {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  text-align: right;
  padding-right: 20px;
}

.contactIcons .btn {
  background-color: #edf0a8;
  border-color: #edf0a8;
}

.contactIcons .btn:hover .fa {
  color: #5d6a77;
}

.contactInfo {
  flex: 1;
  display: flex;
  align-items: center;
}
