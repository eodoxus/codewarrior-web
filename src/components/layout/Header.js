import React, { Component } from "react";
import "./Header.css";

class Header extends Component {
  render() {
    return (
      <div className="Header">
        <header className="Header-header">
          <div id="selfie">
            <img
              src="/sites/default/files/images/green_instgrm.JPG"
              alt="Jason Gordon portrait"
            />
          </div>
          <h1 className="Header-title">Jason Gordon</h1>
        </header>
        <p className="Header-contact-icons">
          <a
            href="https://facebook.com/eodoxus"
            class="btn btn-default btn-sm"
            target="_blank"
            rel="noopener noreferrer"
          >
            <i class="fa fa-facebook" />
          </a>
          <a
            href="https://www.linkedin.com/in/jasongordon"
            class="btn btn-default btn-sm"
            target="_blank"
            rel="noopener noreferrer"
          >
            <i class="fa fa-linkedin" />
          </a>
          <a
            class="btn btn-default btn-sm"
            href="https://github.com/eodoxus"
            target="_blank"
            rel="noopener noreferrer"
          >
            <i class="fa fa-github" />
          </a>
          <a href="mailto:" class="btn btn-default btn-sm">
            <i class="fa fa-envelope" />
          </a>
        </p>
      </div>
    );
  }
}

export default Header;
