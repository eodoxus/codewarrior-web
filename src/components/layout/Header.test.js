import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Header from "./Header";

describe("<Header />", () => {
  let props;
  beforeEach(function() {
    props = {
      avatar: "poo",
      email: "poody",
      home: "/",
      name: "test",
      phone: "2234",
      slogan: "herro"
    };
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Header
        email={props.email}
        name={props.name}
        phone={props.phone}
        portrait={props.avatar}
        slogan={props.slogan}
        url={props.home}
      />,
      div
    );
  });

  it("matches expected snapshot", () => {
    const tree = renderer
      .create(
        <Header
          email={props.email}
          name={props.name}
          phone={props.phone}
          portrait={props.avatar}
          slogan={props.slogan}
          url={props.home}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
