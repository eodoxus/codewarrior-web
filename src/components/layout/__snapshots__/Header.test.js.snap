// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches expected snapshot 1`] = `
<header
  className="header"
>
  <div
    className={undefined}
  >
    <a
      className={undefined}
      href="/"
    >
      <img
        alt="test portrait"
        src="poo"
      />
    </a>
  </div>
  <div
    className={undefined}
  >
    <h1
      className={undefined}
    >
      <a
        href="/"
      >
        test
      </a>
    </h1>
    <h2
      className={undefined}
    >
      herro
    </h2>
  </div>
  <p
    className={undefined}
  >
    <a
      className="btn-sm btn btn-default"
      href="https://facebook.com/eodoxus"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-facebook"
      />
    </a>
    <a
      className="btn-sm btn btn-default"
      href="https://www.linkedin.com/in/jasongordon"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-linkedin"
      />
    </a>
    <a
      className="btn-sm btn btn-default"
      href="https://github.com/eodoxus"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-github"
      />
    </a>
    <span
      className={undefined}
    >
      <a
        className="btn-sm btn btn-default"
        href="tel:2234"
        rel="noopener noreferrer"
        target="_top"
      >
        <i
          className="fa fa-phone"
        />
      </a>
    </span>
    <a
      className="btn-sm btn btn-default"
      href="mailto:poody"
      rel="noopener noreferrer"
      target="_top"
    >
      <i
        className="fa fa-envelope"
      />
    </a>
  </p>
  <p
    className={undefined}
  >
    poody
     
    <br />
    2234
  </p>
</header>
`;
